# Business Undertanding

This is meant to be a fun dataset for you to learn on and use it as something to play around with. This dataset includes data about different species of penguins from different islands with their dimensions and genders. What kind of insights can you get out of this?


In a professional context, the "Business Understanding" phase focuses on defining what success looks like. For this dataset, your objective is to quantify the physical differences between penguin populations to understand how species and environment (islands) influence growth.

Business Goals:
- Identify Patterns: Determine if physical traits (flipper length, beak size, weight) are distinct enough to identify a species without a DNA test.
- Environmental Impact: Assess if the specific island a penguin inhabits correlates with its physical health or size.
- Sexual Dimorphism: Understand the size gap between males and females across different species.


Defintions:
- culmen(ornithology) refers to the upper ridge of a bird’s bill — the top edge running from the base of the beak to the tip.  This feature is often used in bird identification, as the shape (straight, curved up, or down) can distinguish species.


[Data Source](https://www.kaggle.com/datasets/amulyas/penguin-size-dataset)

# **Data Understanding**
- loading
- head and tail( glimpse of how the dat alooks like)
- shape
- info()
- data types
- missing values 
- duplicate
- descritptive stats (categorical)
- descritptive stats (numerical)



#### **Load libraries**

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
penguine_df = pd.read_csv("data/penguins_size.csv")

#### **Head and tail(glimpse of how the data alooks like)**

In [3]:
penguine_df.head()

,species,island,culmen_length_mm,culmen_depth_mm,flipper_length_mm,body_mass_g,sex
0,Adelie,Torgersen,39.1,18.7,181.0,3750.0,MALE
1,Adelie,Torgersen,39.5,17.4,186.0,3800.0,FEMALE
2,Adelie,Torgersen,40.3,18.0,195.0,3250.0,FEMALE
3,Adelie,Torgersen,NaN,NaN,NaN,NaN,NaN
4,Adelie,Torgersen,36.7,19.3,193.0,3450.0,FEMALE


In [4]:
penguine_df.tail()

,species,island,culmen_length_mm,culmen_depth_mm,flipper_length_mm,body_mass_g,sex
339,Gentoo,Biscoe,NaN,NaN,NaN,NaN,NaN
340,Gentoo,Biscoe,46.8,14.3,215.0,4850.0,FEMALE
341,Gentoo,Biscoe,50.4,15.7,222.0,5750.0,MALE
342,Gentoo,Biscoe,45.2,14.8,212.0,5200.0,FEMALE
343,Gentoo,Biscoe,49.9,16.1,213.0,5400.0,MALE


#### **Dimensions**

In [10]:
rows, columns = penguine_df.shape
output = f"RECORDS: {rows} || FEATURES: {columns}"
print(output)

RECORDS: 344 || FEATURES: 7


#### **Info & Data-types**

In [9]:
penguine_df.columns

Index(['species', 'island', 'culmen_length_mm', 'culmen_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex'],
      dtype='str')

In [6]:
penguine_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    str    
 1   island             344 non-null    str    
 2   culmen_length_mm   342 non-null    float64
 3   culmen_depth_mm    342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                334 non-null    str    
dtypes: float64(4), str(3)
memory usage: 18.9 KB


In [8]:
penguine_df.dtypes

species                  str
island                   str
culmen_length_mm     float64
culmen_depth_mm      float64
flipper_length_mm    float64
body_mass_g          float64
sex                      str
dtype: object

#### **Missing values & Duplicates**

In [11]:
penguine_df.isna().sum() # checks for missing values

species               0
island                0
culmen_length_mm      2
culmen_depth_mm       2
flipper_length_mm     2
body_mass_g           2
sex                  10
dtype: int64

In [14]:
penguine_df.duplicated().sum() # checks for duplicate dta and resturns the summ

np.int64(0)

#### **Descritptive stats (categorical & numerical)**

In [16]:
penguine_df.describe() # numerical descriptive stats

,culmen_length_mm,culmen_depth_mm,flipper_length_mm,body_mass_g
count,342.000000,342.000000,342.000000,342.000000
mean,43.921930,17.151170,200.915205,4201.754386
std,5.459584,1.974793,14.061714,801.954536
min,32.100000,13.100000,172.000000,2700.000000
25%,39.225000,15.600000,190.000000,3550.000000
50%,44.450000,17.300000,197.000000,4050.000000
75%,48.500000,18.700000,213.000000,4750.000000
max,59.600000,21.500000,231.000000,6300.000000


Insights: has 344 records and 7 columns
- Data set 
- Presence of missing values in columns (culmen_length_mm -2,culmen_depth_mm-2,flipper_length_mm-2,body_mass_g-2,sex-10)
- Zero duplicate records

culmen_length_mm: 
- 75% of the data were below 48.5mm
- 50% of the data lay about 44.45mm
- 25% of the data lay below 39.225mm
- the elongest length was at 59.6mm
- the smallest was at 32.1mm
- most of the penguines culmen length fell about 43.9219mm
- the spread of the data was quite large as indicated by the standard deviation


In [17]:
penguine_df.describe(include='object') # categorical descriptive stats

/tmp/ipykernel_9477/1013038073.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  penguine_df.describe(include='object') # categorical descriptive stats


,species,island,sex
count,344,344,334
unique,3,3,3
top,Adelie,Biscoe,MALE
freq,152,168,168


In [18]:
penguine_df['sex'].unique() # the unique values in a column

<StringArray>
['MALE', 'FEMALE', nan, '.']
Length: 4, dtype: str

In [19]:
penguine_df['island'].unique() # the unique values in a column

<StringArray>
['Torgersen', 'Biscoe', 'Dream']
Length: 3, dtype: str

In [20]:
penguine_df['species'].unique() # the unique values in a column

<StringArray>
['Adelie', 'Chinstrap', 'Gentoo']
Length: 3, dtype: str